system/Model.php

```diff
@@ -612,13 +581,34 @@ public function insert($data, bool $returnID = true)
 	 * Updates a single record in $this->table. If an object is provided,
 	 * it will attempt to convert it into an array.
 	 *
+	 * updates a single item by id
+	 * $model->update(id, data)
+	 *
+	 * updates multiple items by array of ids
+	 * $model->update(array_id, data)
+	 *
 	 * @param int|string   $id
 	 * @param array|object $data
 	 *
 	 * @return bool
 	 */
-	public function update($id, $data)
+	public function update($id, $data = null)
 	{
+        // because we're going to use "where" for the items to updated
+        // id became a surrogate for data argument
+        // not ideal, but lets see where this goes...
+        if ($data === null)
```

Comment:
The more I think about the polymorphic aspects of our parameters the less I like it. I know we all had a discussion about this previously and decided to go this route, but splitting it into two methods would be more clear. Maybe `updateWith($data)` and `updateById($id, $data)` or something? 

What are your thoughts?

Reply:
I do agree it's not the best solution, I'd really like it to be one method for simplicity sake. However, if we think it's doing too much, we could branch it out. I might be in favor of doing something like `updateWith` since the consistency with delete uses id as its argument. Though we might want to make it (sort of back to the original idea of) `updateWhere` and `deleteWhere` otherwise the update/delete methods use primary keys.